src/cljc/athens/common_events/graph/ops.cljc

```diff
@@ -195,3 +196,39 @@
                            (map :block/new)
                            set)]
     new-uids))
+
+
+(defn structural-diff
+  "Calculates removed and added links (block refs & page links)"
+  [db ops]
+  (let [block-save-ops    (contains-op? ops :block/save)
+        new-blocks        (->> block-save-ops
+                               (map #(select-keys (:op/args %)
+                                                  [:block/uid :block/string])))
+        new-block-uids    (->> new-blocks
+                               (map :block/uid)
+                               set)
+        new-structures    (->> new-blocks
+                               (map :block/string)
+                               (map structure/structure-parser->ast))
+        old-block-strings (->> new-block-uids
```

Comment:
What happens on `:block/save` over a uid that doesn't exist in the db? That should be an invalid operation. But does it crash the runtime or is it handled correctly?

Reply:
then old-block-strings will have `nil` instead of old string